pkg/provider/kubernetes/client_test.go

```diff
@@ -191,6 +194,69 @@ func TestGetSecret(t *testing.T) {
 			},
 			want: []byte(`foobar`),
 		},
+		{
+			name: "successful case with html chars",
+			fields: fields{
+				Client: &fakeClient{
+					t: t,
+					secretMap: map[string]*v1.Secret{
+						"mysec": {
+							Data: map[string][]byte{
+								"html": []byte(`<foobar>`),
+							},
+						},
+					},
+				},
+				Namespace: "default",
+			},
+			ref: esv1beta1.ExternalSecretDataRemoteRef{
+				Key: "mysec",
+			},
+			want: []byte(`{"html":"<foobar>"}`),
+		},
+		{
+			name: "successful case metadata with html special chars and without property",
+			fields: fields{
+				Client: &fakeClient{
+					t: t,
+					secretMap: map[string]*v1.Secret{
+						"mysec": {
+							ObjectMeta: metav1.ObjectMeta{
+								Annotations: map[string]string{"date": "today"},
+								Labels:      map[string]string{"dev": "<seb>"},
+							},
+						},
+					},
+				},
+				Namespace: "default",
+			},
+			ref: esv1beta1.ExternalSecretDataRemoteRef{
+				MetadataPolicy: esv1beta1.ExternalSecretMetadataPolicyFetch,
+				Key:            "mysec",
+			},
+			want: []byte(`{"annotations":{"date":"today"},"labels":{"dev":"<seb>"}}`),
+		},
+		{
+			name: "successful case with binary data",
+			fields: fields{
+				Client: &fakeClient{
+					t: t,
+					secretMap: map[string]*v1.Secret{
+						"mysec": {
+							Data: map[string][]byte{
+								"bindata": binaryTestData,
+							},
+						},
+					},
+				},
+				Namespace: "default",
+			},
+			ref: esv1beta1.ExternalSecretDataRemoteRef{
+				Key:      "mysec",
+				Property: "bindata",
+			},
+			want: []byte(base64.StdEncoding.EncodeToString(binaryTestData)),
```

Comment:
This previously was encoded as string `"\uffed\uffed..."` which was absolutely unusable, see linked issue above.